go/stats/export.go

```diff
@@ -261,15 +228,6 @@ func (v *Duration) String() string {
 	return strconv.FormatInt(int64(v.i.Get()), 10)
 }
 
-// IntFunc converts a function that returns
```

Comment:
> (From @michael-berlin in Reviewable): As @demmer pointed out before, you'll probably need a CounterFunc which will be the renamed version of this.
Note that the removal/rename of this will break the Google internal plugins. Given that, please give me or @alainjobart a head's up before merging this. This way, we can change all internal callers first and verify that they won't break due to this change.

Yes! It's changed to a `GaugeFunc` in https://github.com/vitessio/vitess/pull/3784/commits/a9d7fefbb58f6039dd0852c9531e1d6c3f3cc15f 
I think we'll also need a `CounterFunc`, for the ones that are Counters and not Gauges.

*re: breaking the Google internal plugins*

I'd love to understand how the Google internal plugins use this so that I can avoid breaking it. Are you using the expvar hook or some other way of exporting and converting the expvars to the internal system?

Is it the *removal* here that is likely to break (So, here, we went from `IntFunc` as a parameter to `stats.Publish` to explicitly creating a new `GaugeFunc`), or is it the any/all metric type renames? If it's the latter, head's up that I actually renamed almost all of the metric names. Some of these were to more clearly indicate the Gauge/Counter difference. (Full map in PR description). 

Will that require a lot of work on y'all's part to update the plugins internally?

Reply:
Internal code basically just adds additional stats variables.

Examples:
```go
stats.Publish("XXX", stats.IntFunc(<some method>))

stats.Publish("XXX", stats.FloatFunc(func() float64 { return float64(<some method>) }))

timings:   stats.NewMultiTimings("VtgateXxxApi", []string{"Operation", "Keyspace", "DbType"}),

errorCounters = stats.NewCounters("XXXErrors", "A", "B")

requests = stats.NewMultiCounters("XXX", []string{"A", "B", "C"})
```

As long as it's easy to rewrite this code, it won't be a lot of work and fine.